### PR TITLE
Add support to store/restore/delete binary objects

### LINKED ISSUE #1338: Support storing binary objects
Currently there is no way to store the binary object using NeoFS CLI. This feature can be useful for users which want to store object in container again or restore it after accidental loss.

### PR DESCRIPTION
**Pull request by acid-ant**
Closes #1338

With flag `--binary` user can point to the file with serialized object structure - id + signature + header + payload
Added for development purposes.

Signed-off-by: Anton Nikiforov <an.nikiforov@yadro.com>

### CODE REVIEW

**Comment by cthulhu-rider** [comment]
Do we declare merging behavior of the command? I mean if we provide filepath to serialized object, and specifiy, for example, attributes, they are set in the resulting object.

If so, IMO this complicates the command. I would prefer to have indiviual commands for object formation and storing.

@fyrchik

File: cmd/neofs-cli/modules/object/put.go [hunk 1]
```diff
@@ -117,19 +138,26 @@ func putObject(cmd *cobra.Command, _ []string) {
...
+		prm.SetPayloadReader(payloadReader)
 	} else {
-		fi, err := f.Stat()
-		if err != nil {
-			cmd.PrintErrf("Failed to get file size, progress bar is disabled: %v\n", err)
-			prm.SetPayloadReader(f)
-		} else {
-			p = pb.New64(fi.Size())
+		if binary {
+			p = pb.New64(int64(len(obj.Payload())))
```
Comment (carpawell) [inline cmd/neofs-cli/modules/object/put.go:144]:
```suggestion
			p = pb.New(len(obj.Payload()))
```

File: cmd/neofs-cli/modules/object/put.go [hunk 2]
```diff
@@ -55,22 +56,43 @@ func initObjectPutCmd() {
...
 	pk := key.GetOrGenerate(cmd)
 
 	var ownerID user.ID
-	user.IDFromKey(&ownerID, pk.PublicKey)
-
 	var cnr cid.ID
-	readCID(cmd, &cnr)
 
-	filename := cmd.Flag("file").Value.String()
+	var payloadReader io.Reader
```
Comment (carpawell) [inline cmd/neofs-cli/modules/object/put.go:74]: can we unite the declaration with the definition of the `payloadReader` var?

File: cmd/neofs-cli/modules/object/util.go [hunk 3]
```diff
@@ -23,6 +25,14 @@ const (
 
 	rawFlag     = "raw"
 	rawFlagDesc = "Set raw request option"
+	fileFlag    = "file"
+	binaryFlag  = "binary"
+
+	cidFlag      = "cid"
```
Comment (carpawell) [inline cmd/neofs-cli/modules/object/util.go:31]: maybe share them with `common` pkg? (or via another PR)

File: cmd/neofs-cli/modules/object/put.go (hunk 2 above)
Comment (acid-ant) [reply cmd/neofs-cli/modules/object/put.go]: Refactored

File: cmd/neofs-cli/modules/object/util.go (hunk 3 above)
Comment (acid-ant) [reply cmd/neofs-cli/modules/object/util.go]: Moved into commonflags.

**Review by carpawell — APPROVED** [review]

**Review by fyrchik — APPROVED** [review]

**Comment by fyrchik** [comment]
Could you add `neofs-cli: ` prefix to a commit message?

**Pull request merged**